web/api/v1/api.go

```diff
@@ -1124,29 +1126,30 @@ type tsdbStatus struct {
 	SeriesCountByLabelValuePair []stat `json:"seriesCountByLabelValuePair"`
 }
 
-func (api *API) serveTSDBStatus(r *http.Request) apiFuncResult {
-	db := api.db()
-	if db == nil {
-		return apiFuncResult{nil, &apiError{errorUnavailable, errors.New("TSDB not ready")}, nil, nil}
-	}
-	convert := func(stats []index.Stat) []stat {
-		result := make([]stat, 0, len(stats))
-		for _, item := range stats {
-			item := stat{Name: item.Name, Value: item.Count}
-			result = append(result, item)
-		}
-		return result
+func convertStats(stats []index.Stat) []stat {
+	result := make([]stat, 0, len(stats))
+	for _, item := range stats {
+		item := stat{Name: item.Name, Value: item.Count}
+		result = append(result, item)
 	}
+	return result
+}
 
-	posting := db.Head().PostingsCardinalityStats(model.MetricNameLabel)
-	response := tsdbStatus{
-		SeriesCountByMetricName:     convert(posting.CardinalityMetricsStats),
-		LabelValueCountByLabelName:  convert(posting.CardinalityLabelStats),
-		MemoryInBytesByLabelName:    convert(posting.LabelValueStats),
-		SeriesCountByLabelValuePair: convert(posting.LabelValuePairsStats),
+func (api *API) serveTSDBStatus(*http.Request) apiFuncResult {
+	s, err := api.db.Stats("__name__")
+	if err != nil {
+		if errors.Cause(err) == tsdb.ErrNotReady {
```

Comment:
This is repeated a lot, could we do it in wrap instead?

Reply:
:+1: